diff --git a/Makefile b/Makefile
index afeabb3..33bf011 100644
--- a/Makefile
+++ b/Makefile
@@ -570,6 +570,12 @@ libs-y += $(KCRYPTO_DIR)/
  ZEPHYRINCLUDE += -I$(srctree)/lib/crypto/tinycrypt/include
 endif
 
+ifdef CONFIG_MRAA
+KMRAA_DIR := lib/mraa
+libs-y += ${KMRAA_DIR}/
+ ZEPHYRINCLUDE += -I$(srctree)/lib/mraa/include
+endif
+
 ARCH = $(subst $(DQUOTE),,$(CONFIG_ARCH))
 export ARCH
 ifdef ZEPHYR_GCC_VARIANT
diff --git a/misc/Kconfig b/misc/Kconfig
index 4c05169..1d514eb 100644
--- a/misc/Kconfig
+++ b/misc/Kconfig
@@ -21,6 +21,10 @@ menu "Cryptography"
 source "lib/crypto/tinycrypt/Kconfig"
 endmenu
 
+menu "Mraa"
+source "lib/mraa/Kconfig"
+endmenu
+
 menu "Compile and Link Features"
 
 config  KERNEL_BIN_NAME

